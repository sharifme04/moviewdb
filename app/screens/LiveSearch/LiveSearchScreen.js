import React, {useEffect, useState} from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {
  SafeAreaView,
  FlatList,
  TouchableOpacity,
  TextInput,
  Text,
} from 'react-native';
import {fetchMoviesLive} from '../../actions/movies';
import EmptyComponent from '../../components/EmptyComponent';
import styles from './Style';

const LiveSearchScreen = ({navigation}) => {
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  const [page, setPage] = useState(1);
  const results = useSelector((state) => state.movies?.liveMovies?.results);
  const [total, setTotal] = useState(results);

  useEffect(() => {
    dispatch(fetchMoviesLive(search ? search : ' ', page));
  }, [dispatch, page, search]);

  useEffect(() => {
    if (page === 1) {
      return setTotal(results);
    }
  }, [results, page]);

  const handleLoadMore = () => {
    setPage(page + 1);
    setTotal([...total, ...results]);
  };

  const renderItem = ({item}) => (
    <TouchableOpacity
      style={styles.item}
      onPress={() => navigation.navigate('Details', {id: item.id})}>
      <Text style={styles.title}>{item?.title}</Text>
    </TouchableOpacity>
  );

  return (
    <SafeAreaView>
      <TextInput
        style={styles.container}
        onChangeText={(text) => setSearch(text)}
        placeholder="Please search here"
        value={search}
      />
      <FlatList
        data={total}
        renderItem={renderItem}
        keyExtractor={(item, i) => (item.id + i).toString()}
        ListEmptyComponent={
          <EmptyComponent
            navigation={navigation}
            title="Please write above or"
          />
        }
        onEndReached={handleLoadMore}
        onEndReachedThreshold={0.5}
      />
    </SafeAreaView>
  );
};

export default LiveSearchScreen;
