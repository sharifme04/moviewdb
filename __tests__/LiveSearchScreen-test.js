import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import LiveSearchScreen from '../app/screens/LiveSearch/LiveSearchScreen';
import renderer from 'react-test-renderer';

const mockStore = configureMockStore([thunk]);
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
const store = mockStore({});
const fakeresult = [
  {
    adult: false,
    backdrop_path: '/3eGlcdyxqEN57TtU8ypx52V4JMs.jpg',
    genre_ids: [28, 12, 35, 14, 878],
    id: 98566,
    original_language: 'en',
    original_title: 'Teenage Mutant Ninja Turtles',
    overview:
      "The city needs heroes. Darkness has settled over New York City as Shredder and his evil Foot Clan have an iron grip on everything from the police to the politicians. The future is grim until four unlikely outcast brothers rise from the sewers and discover their destiny as Teenage Mutant Ninja Turtles. The Turtles must work with fearless reporter April and her wise-cracking cameraman Vern Fenwick to save the city and unravel Shredder's diabolical plan.",
    popularity: 112.221,
    poster_path: '/azL2ThbJMIkts3ZMt3j1YgBUeDB.jpg',
    release_date: '2014-08-07',
    title: 'Teenage Mutant Ninja Turtles',
    video: false,
    vote_average: 5.9,
    vote_count: 5078,
  },
];

test('LiveSearchScreen', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <LiveSearchScreen />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('LiveSearchScreen with result', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <LiveSearchScreen total={fakeresult} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('LiveSearchScreen empty result ', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <LiveSearchScreen total={[]} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
