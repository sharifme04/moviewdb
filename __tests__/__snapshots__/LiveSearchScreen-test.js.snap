// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveSearchScreen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Please search here"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 20,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please write above or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please write above or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`LiveSearchScreen empty result  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Please search here"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 20,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please write above or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please write above or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`LiveSearchScreen with result 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Please search here"
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "margin": 20,
        "textAlign": "center",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <RCTScrollView
    ListEmptyComponent={
      <EmptyComponent
        title="Please write above or"
      />
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          Please write above or
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#f77a55",
              "borderRadius": 25,
              "color": "#ffffff",
              "fontSize": 20,
              "opacity": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#f77a55",
                "borderRadius": 25,
                "color": "#ffffff",
                "fontSize": 20,
                "paddingHorizontal": 10,
                "paddingVertical": 5,
              }
            }
          >
            Back To home
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
